File: GmsRcs/src/main/java/com/google/android/gms/rcs/RcsService.java
package com.google.android.gms.rcs;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.util.Log;

import com.google.android.gms.rcs.internal.RcsProvisioningService;

public class RcsService extends Service {
    private static final String TAG = "RcsService";

    @Override
    public IBinder onBind(Intent intent) {
        Log.d(TAG, "RcsService bound: " + intent);
        // Return appropriate binder based on intent action
        if (RcsProvisioningService.SERVICE_INTERFACE.equals(intent.getAction())) {
            return new RcsProvisioningService().getBinder();
        }
        return null;
    }
}

File: play-services-rcs/src/main/java/com/google/android/gms/rcs/RcsService.kt
package com.google.android.gms.rcs

import android.app.Service
import android.content.Intent
import android.os.IBinder
import android.telephony.TelephonyManager
import com.google.android.gms.rcs.internal.IRcsService

class RcsService : Service() {

    private val binder = object : IRcsService.Stub() {
        override fun isRcsEnabled(): Boolean {
            return true
        }

        override fun getRcsConfig(): RcsConfig {
            return RcsConfig.Builder()
                .setEnabled(true)
                .setCapabilities(RcsCapabilities.MESSAGING_INDIVIDUAL)
                .build()
        }

        override fun startRegistration() {
            val tm = getSystemService(TELEPHONY_SERVICE) as TelephonyManager
            val subId = tm.subscriptionId
            // Simulate authentication via SIM
            val authResult = RcsAuthentication.authenticate(subId)
            if (authResult.isSuccess) {
                // Notify RCS backend
                RcsBackend.getInstance(this).register(authResult.identity)
            }
        }

        override fun stopRegistration() {
            RcsBackend.getInstance(this).unregister()
        }
    }

    override fun onBind(intent: Intent?): IBinder? {
        return binder
    }
}